geoapps/io/DirectCurrent/params.py

```diff
@@ -0,0 +1,106 @@
+#  Copyright (c) 2021 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+from __future__ import annotations
+
+from uuid import UUID
+
+from geoapps.io.Inversion import InversionParams
+
+from ..validators import InputValidator
+from .constants import (
+    default_ui_json,
+    forward_defaults,
+    inversion_defaults,
+    required_parameters,
+    validations,
+)
+
+
+class DirectCurrentParams(InversionParams):
+
+    _required_parameters = required_parameters
+    _validations = validations
+    _directive_list = [
+        "UpdateSensitivityWeights",
+        "Update_IRLS",
+        "BetaEstimate_ByEig",
+        "UpdatePreconditioner",
+        "SaveIterationsGeoH5",
+    ]
+
+    def __init__(self, forward=False, **kwargs):
+
+        self.validator: InputValidator = InputValidator(
+            required_parameters, validations
+        )
+        self.inversion_type = "direct_current"
+        self.potential_channel_bool = None
+        self.potential_channel = None
+        self.potential_uncertainty = None
+        self.out_group = None
+
+        self.defaults = forward_defaults if forward else inversion_defaults
+        self.default_ui_json = {k: default_ui_json[k] for k in self.defaults}
+        self.param_names = list(self.default_ui_json.keys())
+
+        for k, v in self.default_ui_json.items():
+            if isinstance(v, dict):
+                field = "value"
+                if "isValue" in v.keys():
+                    if not v["isValue"]:
+                        field = "property"
+                self.default_ui_json[k][field] = self.defaults[k]
+            else:
+                self.default_ui_json[k] = self.defaults[k]
+
```

Comment:
We should probably move this for-loop (and other inversion params) to the base inversion params since it's all the same.

Reply:
done.